examples/websocket-relay/relay-app/src/stores.ts

```diff
@@ -138,58 +150,61 @@ export const nodeStore: Readable<NodeType[]> = derived(
             matchingOneTask.status === "replayed")
         ) {
           const nodeIndex = matchingOneTask.id - 1;
-          const updatedHtml = `${workflowOneNodes[nodeIndex].data.html.slice(
-            0,
-            -2
-          )} style="filter: opacity(75%)" />`;
 
           // Update node in workflow one with opacity to indicate the replayed
           // task
           workflowOneNodes[nodeIndex] = {
             ...workflowOneNodes[nodeIndex],
-            data: { html: updatedHtml },
           };
 
           // Skip adding new nodes to workflow two
           return nodes;
         }
 
+        const id = String(index + idOffset);
+
         // @ts-ignore
         nodes = [
           ...nodes,
           {
-            id: String(index + idOffset),
-            position: { x: 500 + (index + 1) * 250, y: 450 },
-            data: {
-              html:
-                task.status === "replayed"
-                  ? `<img src="data:image/png;base64,${task.receipt?.out[1]}" draggable="false" style="filter: opacity(75%)" />`
-                  : `<img src="data:image/png;base64,${task.receipt?.out[1]}" draggable="false" />`,
-            },
-            width: 150,
-            height: 150,
-            bgColor: "white",
-            borderColor: "transparent",
+            id,
+            position: { x: 208 + (index + 1) * 275, y: 570 },
+            task,
           },
         ];
       }
+
       return nodes;
     }, []);
 
+    // Nodes don't always have the correct task receipts because activeWorkflow.step can get out of sync
+    // when the BE returns two messages at the same time, which causes the wrong receipt to be associated
+    // with a task. So here we ensure the correct receipt is assigned to the correct node and task by
+    // checking the NODE_TASK_MAP, which states the correct sequence of ops based on a node.id
+    const workflowNodes = [...workflowOneNodes, ...workflowTwoNodes].map(
+      (node) =>
+        NODE_TASK_MAP[node.id] !== node?.task?.receipt?.meta?.op
+          ? {
+              ...node,
+              task: {
+                ...(taskStore["one"].find(
```

Comment:
I had to explicitly set `taskStore['one']` and `taskStore['two']` here because `taskStore[node.task.workflowId]` would fail if the task hadn't already been executed for that workflow. it's not pretty and there are maybe better ways to do it, but it felt best to keep all this logic in one place with a single disclaimer comment above it 🤷🏼

Reply:
Yep, that works. We can always refactor it later if we think of another idea. 👍